server/configuration.go

```diff
@@ -136,15 +137,15 @@ func (c *jitsisettings) IsValid() error {
 
 	if c.UseJaaS {
 		if len(c.JaaSApiKey) == 0 {
-			return fmt.Errorf("error no JaaS Api Key was provided for JaaS")
+			mlog.Error("error no JaaS Api Key was provided for JaaS")
```

Comment:
Usually, for logging errors we use the plugin API. Also, why not returning the error in these cases? Is there any situation where UseJaaS can be used without these values?

Reply:
If JaaS is enabled, plugin will fail to activate - OnActivate will return error in case JaaSApiKey, JaaSAppID or JaaSPrivateKey is not specified. This might have happened because i already had the plugin installed with JaaS enabled but somehow these settings were erased...(i may have reconfigured Mattermost or clear cache somehow). This is why i removed the error.

Reply:
Ok, I think I get it. This is needed here since in order to see the custom settings, the plugin must be activated. Am I right? If that is the case, I wonder if we should modify the rest of the is valid function.

Reply:
I have to look more into it, it happened right after uploading the plugin - it failed to activate(tried to enable the plugin) and JaaS was already enabled but the JaaSApiKey, JaaSAppID or JaaSPrivateKey were not set.